sdk/messaging/azeventhubs/internal/links.go

```diff
@@ -274,16 +336,18 @@ func (l *Links[LinkT]) newLinkState(ctx context.Context, partitionID string) (*l
 	tmpLink, err := l.newLinkFn(ctx, session, l.entityPathFn(partitionID))
 
 	if err != nil {
+		azlog.Writef(exported.EventConn, "(%s): Failed to create link for partition ID '%s': %s", ls.String(), partitionID, err)
 		_ = ls.Close(ctx)
```

Comment:
And here we _don't_ overwrite the last error?

Reply:
Right, I think the non-overwrite behavior is better. Leaving this one, changing the other to match the perfect model this one has provided.